nixos/modules/services/misc/gitea.nix

```diff
@@ -365,6 +365,8 @@ in
     ];
 
     services.gitea.settings = {
+      "cron.update_checker".ENABLED = false;
+
```

Comment:
Can you please put this in `options.services.gitea.settings` as a new option with `default = false;` instead?

Reply:
or `lib.mkDefault` this option.

Reply:
Why? We normally disable updaters no questions asked sometimes even completely patching them out.

Reply:
Okay, sorry.

I'll try to outline my original thought process.
But please free to just dismiss this thread, if you don't feel like it.

That said, a few things came into play here:
1. I was originally under the impression that the reasoning behind disabling an updater is slightly more nuanced.
   Literal *updater*, that try to -- well -- *update* the binary itself, that are obviously broken, versus update checker/notifier.
   I agree that binary updater should be disabled, as you phrase it, "no questions asked".
   But I do see value in keeping update notifications enabled.
   Depending on how they are implemented, they might allow software maintainers to get a rough idea how many use their software.
   In case of gitea/forgejo this is even pretty anonymous.
   And, even more importantly, the update notification only shows up in the admin dashboard and also just links to the blog[^1], instead of providing wrong update instructions.
   From what I can tell, GitLab's update notification/badge isn't disabled either.
   Same applies for Nextcloud, iirc: update notification enabled, updater disabled.
   Looking at
    - https://github.com/NixOS/nixpkgs/commit/cd80cab3b058c9a4de6c59e91177aa6a1b9c1eca
      Its update notification seems to be much more pushy and also misleading[^2]
    - https://github.com/NixOS/nixpkgs/pull/34127
      Broken update instructions
    - https://github.com/NixOS/nixpkgs/pull/166880
      Broken update instructions and tracking
1. I could image benefits where an update isn't picked up by @r-ryantm and a nixpkgs maintainer might miss it otherwise. Non-pushy update notification provide a way to close that loop and may even introduce new maintainers to nixpkgs or the package itself. This doesn't apply to gitea though, as enough have their eyes on it already (including you!).
1. I thought it would make a good example for novice nix users on how the nix attrs -> `.ini` (with a dot in the section) mapping works. Turns out I missed that `options.services.gitea.settings` already provides such example[^3]. If I hadn't missed that, I would have suggested `lib.mkDefault`.
1. Personally, I want to keep it enabled on my instance, and having to use `lib.mkForce` makes it slightly more inconvenient /shrug

But honestly, you have obviously much better overview of nixpkgs than me.
Those are just my thoughts as a fairly new person in nixpkgs.

And maybe there is some guideline in a section in the manual or issue/forum thread about this already, which I couldn't find, rendering this complete redundant :eyes: 

[^1]: https://github.com/go-gitea/gitea/blob/d4f35bd681af0632da988e15306f330e020422b2/options/locale/locale_en-US.ini#L2514
[^2]: https://github.com/gpodder/gpodder/blob/7323e0c604392879b5d57dc7623583961a8b0fe1/src/gpodder/gtkui/main.py#L3774-L3785
[^3]: https://github.com/NixOS/nixpkgs/blob/276bb35d857b730903127fe5e32a1ce4c6d9b2c6/nixos/modules/services/misc/gitea.nix#L272-L276

Reply:
> 1. Depending on how they are implemented, they might allow software maintainers to get a rough idea how many use their software.

We often disable such data collections in nixpkgs, too as they are more often than not missing the appropriate privacy policies, similar to DFSG from Debian.

Also the notification is for most people none actionable.



> I agree with your reasoning, hence, I am in favor of mkDefault.

I don't really care but if it makes you happy I add that.